components/RemediesList.js

```diff
@@ -2,12 +2,31 @@ import Image from "next/image";
 import Link from "next/link";
 import styled from "styled-components";
 import FavoriteButton from "./FavoriteButton";
+import SearchBar from "./SearchBar";
+import { useState } from "react";
 
 export default function RemediesList({ remedies, handleToggleFavorite }) {
+  const [searchQuery, setSearchQuery] = useState("");
+  const filteredRemedies = remedies.filter((remedy) =>
+    remedy.title.toLowerCase().includes(searchQuery.toLowerCase())
+  );
```

Comment:
This function filters remedies by checking if the `searchQuery` is found **anywhere in the remedy's title**. 

This approach finds matches anywhere in the title but prioritises results that start with the search term, which usually leads to more relevant matches. 

```suggestion
 const filteredRemedies = remedies.filter((remedy) => {
  const title = remedy.title.toLowerCase();
  const lowerCaseQuery = searchQuery.toLowerCase();
  return title.includes(lowerCaseQuery) || title.startsWith(lowerCaseQuery);
});
```